dataset-workflows/simple_s3_ingest/cdk/cdk_stack.py

```diff
@@ -0,0 +1,180 @@
+            runtime=aws_lambda.Runtime.FROM_IMAGE,
+            memory_size=4096,
+            timeout=core.Duration.seconds(60)
+        )
+        generate_stac_item_lambda.add_to_role_policy(full_bucket_access)
+
+        db_write_role = aws_iam.Role(
+            self,
+            "PGStacLoaderRole",
+            assumed_by=aws_iam.ServicePrincipal("lambda.amazonaws.com"),
+            managed_policies=[
+                aws_iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "service-role/AWSLambdaBasicExecutionRole"
+                )
+            ],
+        )
+
+
+        db_write_lambda = aws_lambda.Function(
+            self,
+            f"{id}-{name}-write-db-fn",
+            role=db_write_role,
+            code=aws_lambda.Code.from_asset_image(
+                directory="../../lambdas/db-write",
+                file="Dockerfile",
+                entrypoint=["/usr/local/bin/python", "-m", "awslambdaric"],
+                cmd=["handler.handler"],
+            ),
+            handler=aws_lambda.Handler.FROM_IMAGE,
+            runtime=aws_lambda.Runtime.FROM_IMAGE,
+            memory_size=4096,
+            timeout=core.Duration.seconds(360),
+            environment=dict(
+                STAC_DB_HOST=os.environ["STAC_DB_HOST"],
+                STAC_DB_USER=os.environ["STAC_DB_USER"],
+                # TODO: Aimee - may want to be consistent about environment name for PG PASSWORD.
+                # PGPASSWORD is the name of the variable used by the psql cli to load without requesting a password via a command line prompt.
+                PGPASSWORD=os.environ["PGPASSWORD"],
+            ),
+            vpc=database_vpc,
+            # vpc_subnets=ec2.SubnetSelection(subnet_type=ec2.SubnetType.PRIVATE),
+            security_groups=[lambda_function_security_group],
+        )
+
+        db_write_lambda.add_to_role_policy(ec2_network_access)
+
+        s3_start_state = stepfunctions.Pass(self, "S3 Discovery StartState")
+        s3_discover_task = tasks.LambdaInvoke(
+            self, "S3 Discover Task", lambda_function=s3_discovery_lambda
+        )
+
+        s3_generate_stac_item_task = tasks.LambdaInvoke(
+            self,
+            "S3 Generate STAC Item Task",
+            lambda_function=generate_stac_item_lambda,
+        )
+
+        s3_db_write_task = tasks.LambdaInvoke(
+            self,
+            "S3 DB Write task",
+            lambda_function=db_write_lambda,
+            input_path="$.Payload",
+        )
+
+        map_stac_items = stepfunctions.Map(
+            self,
+            "Map STAC Item Generator",
+            max_concurrency=10,
+            items_path=stepfunctions.JsonPath.string_at("$.Payload"),
+        )
+
+        # Create stac item for each element
+        map_stac_items.iterator(s3_generate_stac_item_task.next(s3_db_write_task))
+
+        s3_wflow_definition = s3_start_state.next(s3_discover_task).next(map_stac_items)
```

Comment:
Just to make sure I understand, there is not branching logic given the state machine input _yet_ right?